### PR TITLE
Added the notification for the third-party civilizations

### PR DESCRIPTION
**Pull request by lyrjie**
If a civilizations knows about some two civilizations, it will be notified when:
* They declare the war
* They make a peace
* They sign the declaration of peace
* One denounce the other

Additionally, when civ enters a new era, all civs that know about it get a notification

Suggested by @TheBucketeer

era
war

### CODE REVIEW

File: core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt [hunk 1]
```diff
@@ -415,8 +415,14 @@ class DiplomacyManager() {
     fun makePeace(){
         diplomaticStatus= DiplomaticStatus.Peace
 
-        getCommonKnownCivs().forEach {
-            it.addNotification("[${civInfo.civName}] and [${otherCiv().civName}] have signed the Peace Treaty!", null, Color.WHITE)
+        if (otherCiv().isAtWarWith(civInfo)) {
```
Comment (lyrjie) [inline core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt]: Couldn't think of a cleaner way to do it

File: android/assets/jsons/translationsByLanguage/template.properties [hunk 2]
```diff
@@ -314,6 +314,11 @@ Indeed! =
 Denounce ([numberOfTurns] turns) = 
 We will remember this. = 
 
+[civName] has declared war on [targetCivName]! = 
+[civName] and [targetCivName] have signed the Peace Treaty! = 
+[civName] and [targetCivName] have signed the Declaration of Friendship! = 
+[civName] has denounced [targetCivName]!" =
```
Comment (r3versi) [inline android/assets/jsons/translationsByLanguage/template.properties]: Typo: there is a wild `"`

File: core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt (hunk 1 above)
Comment (r3versi) [reply core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt]: What's the point in checking `otherCiv().isAtWarWith(civInfo)`? If you are in `makePeace` it means a trade with a peace treaty has been signed, so the two civs must be at war.

Comment (lyrjie) [reply core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt]: When AI makes peace with AI, there's this piece of code
```
        transferTrade(ourCivilization, otherCivilization, currentTrade)
        transferTrade(otherCivilization, ourCivilization, currentTrade.reverse())
```
which triggers the `makePeace()` twice.
I check the `otherCiv`s diplomacy manager to see if their side of peace deal is already done, and if it is, I don't push notification.
This avoids double-notification like "Russian made peace with America" and "America made peace with Russia"

File: android/assets/jsons/translationsByLanguage/template.properties (hunk 2 above)
Comment (lyrjie) [reply android/assets/jsons/translationsByLanguage/template.properties]: Thanks for a catch

**Comment by lyrjie** [comment]
Added some tests. Let me know if they are poorly written / useless

**Comment by yairm210** [comment]
Wow I was not expecting this level of testing
Wow

**Pull request merged**